query/src/main/java/org/infinispan/query/impl/SearchManagerImpl.java

```diff
@@ -148,8 +146,7 @@ public void purge(Class<?> entityType) {
 
    @Override
    public <T> T unwrap(Class<T> cls) {
-      if (SearchIntegrator.class.isAssignableFrom(cls) || ExtendedSearchIntegrator.class.isAssignableFrom(cls)
-            || SearchFactoryState.class.isAssignableFrom(cls)) {
+      if (SearchIntegrator.class.isAssignableFrom(cls)) {
```

Comment:
It won't work in OSGi. For experimenting IMO snapshot dependencies should be used.